### PR TITLE
[tensorboard] Clarify API and add examples for all methods

### PR DESCRIPTION
**Pull request by lanpa**
As a part of supporting writing data into TensorBoard readable format, we show more example on how to use the function in addition to the API docs.

### CODE REVIEW

**Review by orionr — CHANGES_REQUESTED** [review]
Great idea! However, let's use non-dummy data for this. You can start with some of the colab notebook info I sent. Ultimately, though, this might be more complex and so will need to actually go in our PyTorch tutorials.

One request would be to add simple examples (you have some great ones already) to all of the `SummaryWriter.add_*` methods.

File: torch/utils/tensorboard/writer.py [hunk 1]
```diff
@@ -235,7 +252,7 @@ def _get_file_writer(self):
             else:
                 self.file_writer = FileWriter(logdir=self.log_dir, **self.kwargs)
             self.all_writers = {self.file_writer.get_logdir(): self.file_writer}
-        return self.file_writer
+        return self.file_writer
```
Comment (orionr) [inline torch/utils/tensorboard/writer.py]: Nit - can you remove this whitespace?

File: torch/utils/tensorboard/writer.py [hunk 2]
```diff
@@ -246,6 +263,21 @@ def add_scalar(self, tag, scalar_value, global_step=None, walltime=None):
...
+            x = range(100)
+            for i in x:
+                writer.add_scalar('y=2x', i * 2, i)
+            writer.close()
+
+        Expected result:
+
+        .. image:: _static/img/tensorboard/add_scalar.png
+           :scale: 50 %
+
```
Comment (orionr) [inline torch/utils/tensorboard/writer.py]: For add_scalars below can we also add the preample

```
            from torch.utils.tensorboard import SummaryWriter
            writer = SummaryWriter()
```

?

File: torch/utils/tensorboard/writer.py [hunk 3]
```diff
@@ -601,6 +694,10 @@ def add_embedding(self, mat, metadata=None, label_img=None, global_step=None, ta
     def add_pr_curve(self, tag, labels, predictions, global_step=None,
                      num_thresholds=127, weights=None, walltime=None):
         """Adds precision recall curve.
+        Precision recall curve lets you understand you model's performance under different
```
Comment (orionr) [inline torch/utils/tensorboard/writer.py]: Nit - add a newline above this line. Also, can we change the text to

```
Plotting a precision-recall curve lets you understand you model's performance under different
threshold settings. With this function, you provide the ground truth labeling (T/F) and
prediction confidence (usually the output of your model) for each target. The TensorBoard UI
will let you choose the threshold interactively.
```

File: torch/utils/tensorboard/writer.py [hunk 4]
```diff
@@ -352,6 +400,31 @@ def add_image(self, tag, img_tensor, global_step=None, walltime=None, dataformat
             convert a batch of tensor into 3xHxW format or call ``add_images`` and let us do the job.
             Tensor with :math:`(1, H, W)`, :math:`(H, W)`, :math:`(H, W, 3)` is also suitible as long as
             corresponding ``dataformats`` argument is passed. e.g. CHW, HWC, HW.
+
+        Examples::
+
+            from torch.utils.tensorboard import SummaryWriter
+            import numpy as np
+            my_img = np.zeros((3, 100, 100))
```
Comment (orionr) [inline torch/utils/tensorboard/writer.py]: Nit: Can we change variable names from `my_img` to `img` and from `my_img_HWC` to `img_hwc`?

**Review by orionr — COMMENTED** [review]
A few comments to address. This is looking great. Once you're done with these I'll run the doc build locally and see how it all looks on the page.

File: torch/utils/tensorboard/writer.py (hunk 1 above)
Comment (lanpa) [reply torch/utils/tensorboard/writer.py]: Interest, flake8 didn't complain about it.

File: torch/utils/tensorboard/writer.py (hunk 2 above)
Comment (lanpa) [reply torch/utils/tensorboard/writer.py]: also added add_scalars to tensorboard.rst

File: torch/utils/tensorboard/writer.py (hunk 4 above)
Comment (lanpa) [reply torch/utils/tensorboard/writer.py]: also fixed similar issue in `add_images`

**Review by orionr — APPROVED** [review]

**Comment by lanpa** [comment]
Thanks!

**Pull request closed without merging**